### PR TITLE
Azure Pipelines as an additional CI tool

### PR DESCRIPTION
**Pull request by janpio**
This PR creates a minimal configuration file for Azure Pipelines of Azure DevOps.

More information about Azure Pipelines here:
https://azure.microsoft.com/en-us/blog/announcing-azure-pipelines-with-unlimited-ci-cd-minutes-for-open-source/
https://azure.microsoft.com/en-us/services/devops/pipelines/

For this to actually do anything, an Apache INFRA issue has to be processed. I will update the PR when/if this happened.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    vmImage: 'Ubuntu 16.04'
+  strategy:
+    maxParallel: 3
+    matrix:
+      node_6_x:
+        node_version: 6.x
+      node_8_x:
+        node_version: 8.x
+      node_10_x:
+        node_version: 10.x
```
[the remark this answers is not in the document]
Comment (janpio) [reply azure-pipelines.yml]: That works indeed: https://janpio.visualstudio.com/cordova-lib/_build/results?buildId=67&view=logs Changed here.

[the remark this answers is not in the document]
Comment (raphinesse) [reply azure-pipelines.yml]: Nice!

**Comment by raphinesse** [comment]
Now why do we have four checks on this PR again? That only happens sometimes...

**Comment by janpio** [comment]
Because I created the branch on the `apache/cordova-lib` repo, and not my branch. Both Travis and AppVeyor are probably configured to build both a PR (merge of a branch into `master`) and a branch itself.

**Comment by raphinesse** [comment]
Aaaaaah. Yes. Thanks for explaining it to me 😅

**Comment by janpio** [comment]
(To add to the confusion: One of those Appveyor builds runs in the `ApacheSoftwareFoundation` account, the other in `Humbedooh`. Both have crazy queue times. Yay.)

**Comment by kaylangan** [comment]
I'm a Program Manager on the Azure Pipelines team. Let me know if you have any questions or suggestions. I'm happy to help!

**Comment by janpio** [comment]
Hi @kaylangan - you will probably want to follow/watch https://issues.apache.org/jira/projects/INFRA/issues/INFRA-17030 then, this will decide if we can use Azure Pipelines from a Apache/"legal" perspective. 

From a more technical view, it would be awesome if someone could look at https://stackoverflow.com/questions/52374098/how-to-report-individual-jobs-of-pipeline-as-github-checks-for-azure-pipelines and reply there. All the rest was pretty simple, quick and painless 👍

**Comment by DavidStaheli** [comment]
Hi @janpio.  I work with @kaylangan.  We're adding support for showing individual jobs in GitHub Checks right now (our current 3-week sprint).  It should be in production in October 2018.  Any time in the future, we appreciate any input you have about how Azure Pipelines can better meet your needs.

**Comment by kaylangan** [comment]
> Hi @kaylangan - you will probably want to follow/watch https://issues.apache.org/jira/projects/INFRA/issues/INFRA-17030 then, this will decide if we can use Azure Pipelines from a Apache/"legal" perspective.

Please let us know if you run into any blockers around permissions that we may be able to help with!

**Comment by janpio** [comment]
I extracted the steps to their own "template" file to avoid the repetition (and added conditions as we run different commands on Linux vs. Windows). I also added macOS as an operating system - because why not.

After talking to some Azure Devops Pipelines people I have a Pipeline set up where we can manually trigger CI runs of this repository (no CI on PRs, commits, tags etc, also no check here in GitHub). This is the latest commit: https://dev.azure.com/apache-cordova/cordova-lib/_build/results?buildId=4&view=logs

**Comment by raphinesse** [comment]
I will close this in favor of implementing GitHub Actions. Thanks for all your work on this, Jan!

If you disagree, please do not hesitate to reopen this PR.

**Comment by janpio** [comment]
Nah, GH Actions are a newer version of Azure Pipelines anyway so this makes total sense. Good luck!

**Pull request closed without merging**